seed/tasks.py

```diff
@@ -863,27 +864,42 @@ def _finish_matching(import_file, progress_key):
     import_file.save()
     cache.set(progress_key, 100)
 
+
 def _normalize_address_str(address_val):
-    """Normalize the address to conform to short abbreviations."""
-    #if this string is empty the regular expression in the sa wont like it, and fail, so leave returning nothing
+    """
+    Normalize the address to conform to short abbreviations.
+
+    If an invalid address_val is provided, None is returned.
+
+    If a valid address is provided, a normalized version is returned.
+    """
+
+    # if this string is empty the regular expression in the sa wont
+    # like it, and fail, so leave returning nothing
     if not address_val:
-        return
-    # so the parse function uses a regular expression defined in Regexes to make sure that the address conforms to a standard format, 
-    # the function does not recognize an address unless it has more than just  an address line, so this hack below adds a fake post code, 
-    # a more elegant solution would be to  change the regular expression so that a simple address is ok
-    address_val=address_val +', 95472'
-    #now parse the address into number, street name and street type, and normalize all street types to be
-    addr=sa.parse(address_val)
+        return None
+
+    # now parse the address into number, street name and street type,
+    # and normalize all street types to be
+    parser = StreetAddressParser()
+    addr = parser.parse(str(address_val))  # TODO: should probably use unicode()
+    normalized_address = ''
+
     if not addr:
-        return
-    if addr.has_key('number'): 
-        NormalizedAddress=addr['number']
-    if addr.has_key('street'): 
-        NormalizedAddress=NormalizedAddress+' '+addr['street']  
-    if addr.has_key('type'):
-        NormalizedAddress=NormalizedAddress+' '+addr['type']    
-    return NormalizedAddress.lower()
-    
+        return None
+
+    if 'house' in addr and addr['house'] is not None:
+        normalized_address = addr['house']
+
+    if 'street_name' in addr and addr['street_name'] is not None:
+        normalized_address = normalized_address + ' ' + addr['street_name']
+
+    if 'street_type' in addr and addr['street_type'] is not None:
+        normalized_address = normalized_address + ' ' + addr['street_type']
+
+    return normalized_address.lower().strip()
```

Comment:
The same library has an address formatter module that converts street type abbreviations (Street ==> St).

Instead of normalized_address.lower().strip()
We can do:
addr_formatter = StreetAddressFormatter()
addr_formatter.abbrev_street_avenue_etc(normalized_address)

[Full documentation for street-address](https://github.com/pnpnpn/street-address)